src/lib.rs

```diff
@@ -140,6 +140,8 @@ pub fn parse_and_resolve(filename: &str, cache: &mut FileCache, target: Target)
         }
         Ok(file) => {
             sema::sema(file, cache, &mut ns);
+            sema::unused_variable::check_unused_namespace_variables(&mut ns);
+            sema::unused_variable::check_unused_events(&mut ns);
         }
```

Comment:
It is better if `sema::sema()` calls the unused variable checks. I realise that `sema::sema()` is recursive, and you don't want this to be called for every recursive call. So maybe rename `sema::sema_file()` or something similar and create a new function `sema::sema()` which calls the unused variable checks.

Reply:
Fixed